cmd/lifecycle/analyzer.go

```diff
@@ -207,21 +220,29 @@ func (aa analyzeArgs) analyze() (platform.AnalyzedMetadata, error) {
 	return analyzedMD, nil
 }
 
-func (a *analyzeCmd) registryImages() []string {
-	var registryImages []string
-	if a.platform06.cacheImageTag != "" {
-		registryImages = append(registryImages, a.platform06.cacheImageTag)
-	}
-	if !a.useDaemon {
-		registryImages = append(registryImages, append([]string{a.imageName, a.previousImage, a.runImageRef}, a.additionalTags...)...)
-	}
-	return registryImages
+func (a *analyzeCmd) platformAPIVersionGreaterThan06() bool {
+	return api.MustParse(a.platform.API()).Compare(api.MustParse("0.7")) >= 0
 }
 
 func (a *analyzeCmd) restoresLayerMetadata() bool {
 	return !a.platformAPIVersionGreaterThan06()
 }
 
-func (a *analyzeCmd) platformAPIVersionGreaterThan06() bool {
-	return api.MustParse(a.platform.API()).Compare(api.MustParse("0.7")) >= 0
+func (a *analyzeCmd) ReadableImages() []string {
+	var readableImages []string
+	if a.platformAPIVersionGreaterThan06() {
```

Comment:
I think it would make more sense to have the platform version guarding the call to `VerifyRegistryAccess`. Conceivably, we might want to know the readable registry images for earlier platform api (see https://github.com/buildpacks/lifecycle/pull/646/files#r666182517).

Reply:
The `VerifyRegistryAccess` is on the `image` package, I will need to expose the value in the interface but I do not like that approach. I can move the validation like this 

```
func (a *analyzeCmd) Privileges() error {
        // code
	if a.platformAPIVersionGreaterThan06() {
		if err := image.VerifyRegistryAccess(a, a.keychain); err != nil {
			return cmd.FailErr(err)
		}
	}
         // code
}
```

Reply:
That looks great to me!

Reply:
Fixed in the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/5c900400a2e89ae0d9f8a8d0c02243dec574ceba)